docs/types.rst

```diff
@@ -266,6 +266,128 @@ to and from all integer types but implicit conversion is not allowed.
         }
     }
 
+.. index:: ! function type, ! type; function
+
+.. _function_types:
+
+Function Types
+--------------
+
+Function types can be used to assign functions to variables and passing them
```

Comment:
`passing them to` and `return them from` do not balance.  I think it's either `pass them to and return them from` or `passing them to and returning them from`.